AI/modules/data_collector/company_fundamentals_data.py

```
@@ -78,58 +71,79 @@ def fetch_and_calculate_metrics(self, ticker: str):
+        return processed_data
 
+    # ==========================================
+    # 💎 엔진 2: FMP API (과거 10년 치 고급 수집)
+    # ==========================================
+    def fetch_fmp_metrics(self, ticker: str) -> List[tuple]:
+        if not self.api_key: return []
+        base_url = "https://financialmodelingprep.com/api/v3"
+        limit = 10
+        
+        try:
+            inc_resp = requests.get(f"{base_url}/income-statement/{ticker}?limit={limit}&apikey={self.api_key}")
```

Ruff (0.15.5)

[error] Probable use of `requests` call without timeout

(S113)